_posts/2022-07-24-opensearch-plugin-zips-now-in-maven-repo.markdown

```diff
@@ -0,0 +1,74 @@
+
+### Maven publish task with custom gradle plugin.
+
+The publishing of zips to central maven repo had been possible with custom gradle plugin `opensearch.pluginzip`.
+Starting with release `2.1.0` all OpenSearch gradle supported plugins will have custom gradle plugin `opensearch.pluginzip` applied to the gradle project, this will create a new task `publishPluginZipPublicationToZipStagingRepository` which does all the heavy lifting in identifying the distribution plugin zip, setting the maven coordinates, generating the POM file, updating the user passed POM fields and publishing it to maven repo. More details and inner workings of the plugin can found at [opensearch-plugins](https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#opensearchpluginzip).
+
+
+**Figure 1**: Workflow that ship generated plugin zips to maven
+
+![Figure 1: Workflow that ship zips to maven]({{ site.baseurl }}/assets/media/blog-images/2022-07-24-opensearch-plugin-zips-now-in-maven-repo/figure1.png){: .img-fluid }
+
+### Benefits of plugin zips in maven:
+
+* The plugin dependencies can now be fetched using the right maven coordinates from maven repo and a user need not use a zip of some random version built on a developer desktop.
+* The development dependent `SNAPSHOT` version zips can be directly fetched from maven using the dependency mechanism.
+* Not required to check in the zips manually to any `src/` folders, as zips can now be fetched using the right `grounID`, `artifactID` and `version`.
+* The published zips to central maven repo are signed with `.asc`, `.md5`, `.sha1`, `.sha256`, `.sha512` extensions.
+* The tests and CI workflows can directly run against the zips fetched from maven repo, instead of requiring a manual download and checking into the project.
+
+### Consume plugin zips from maven:
+
+**Using mvn cli:**
+
+Consume from Central Maven repo:
+```
+mvn dependency:get -DgroupId=org.opensearch.plugin -DartifactId=opensearch-job-scheduler -Dversion=2.1.0.0 -Dpackaging=zip -DremoteRepositories=central::default::https://repo.maven.apache.org/maven2,myrepo::::http://myrepo.com/maven2
+```
+Consume from Snapshot Maven repo:
+```
+Snapshot Maven repo
+mvn dependency:get -DgroupId=org.opensearch.plugin -DartifactId=opensearch-job-scheduler -Dversion=2.1.0.0-SNAPSHOT -Dpackaging=zip -DremoteRepositories=https://aws.oss.sonatype.org/content/repositories/snapshots/
+```
+
+**Gradle Project: using build.gradle file**
+
+```
+dependencies {
+    classpath "org.opensearch.plugin:opensearch-job-scheduler:2.1.0.0@zip"
+    classpath "org.opensearch.plugin:opensearch-knn:2.1.0.0@zip"
+}
+```
+
+**Maven Project: using pom.xml file**
+
+```
+<dependencies>
+        <dependency>
+            <groupId>org.opensearch.plugin</groupId>
+            <artifactId>opensearch-job-scheduler</artifactId>
+            <version>2.1.0.0</version>
+            <packaging>zip</packaging>
+        </dependency>
+</dependencies>
+```
+
+### Summary:
```

Comment:
```suggestion

```

Reply:
Small change with the command

Consume from Central Maven repo:
```
mvn dependency:get -DgroupId=org.opensearch.plugin -DartifactId=opensearch-job-scheduler -Dversion=2.1.0.0 -Dpackaging=zip -DremoteRepositories=https://repo1.maven.org/maven2
```